Issac's branch

IsaacSo919 opened the pull request:
The first push request

IsaacSo919 committed:
TEST 1

TEST 1

IsaacSo919 committed:
TEST 1

TEST 1

IsaacSo919 committed:
TEST 1

TEST 1

IsaacSo919 committed:
asdasd

IsaacSo919 committed:
carbon value fixed between zero and ten

carbon value fixed between zero and ten

IsaacSo919 committed:
Update the tree with a startup container and text and new icon button

Update the tree with a startup container and text and new icon button

IsaacSo919 committed:
QR Code scanner fixed, Tree system added

QR Code scanner fixed, Tree system added

IsaacSo919 committed:
Newest and last

IsaacSo919 committed:
Merge branch 'develop' into Issac's-branch

IsaacSo919 committed:
fetched with the develop branch

fetched with the develop branch

Yout-bit commented on src/gseproject/EXEplorer/views.py:
```diff
@@ -85,25 +85,56 @@ def get_carbon_footprint_value(request):
+        # Check if the carbon footprint goes below 0
+        if user_profile.carbonFootprint < 0:
+            user_profile.carbonFootprint = 0
+
```
Can we combine the whole abouve section into something like:
```
if request.method == 'POST':
        user_profile = request.user.userprofile

        if user_profile.carbonFootprint < 10:
            user_profile.carbonFootprint += 1
```
do we need the below 0 part if we are only incrementing the value?

IsaacSo919 replied:
It is designed for preventing the value from dropping below 0 in my design, and I got a function to increase the value for an interval

Yout-bit replied:
now that you have added the decrease function it makes more sense, however i still would prefer only the increment function to check the max value and only the decrease to check the min

Yout-bit replied:
Also this is another case of repeated code.

IsaacSo919 replied:
sorted

Yout-bit commented on src/gseproject/templates/game/QR code.html:
```diff
@@ -16,9 +16,10 @@
-
-<script>
-    function domReady(fn) {
+<script src="path/to/base_game_template.js"></script>
```
base_template.js should already be imported, making the line `<script src="path/to/base_game_template.js"></script>` redundant

IsaacSo919 replied:
ok, should i delete that?

Yout-bit replied:
yeah just drop that line

Yout-bit commented on src/gseproject/templates/game/tree.html:
```diff
@@ -44,85 +86,73 @@
             timerElement.textContent = ''; // Clear timer display
+            if (currentStage === stages.length) {
+                incrementCherriesValue(); // Increment cherries value when tree is fully grown
+                alert('Congratulations! You have grown a full tree and earned 10 bonus cherries!, carbon footprint reduced by 1!');
```
It would be nicer if this text apperead in a pop up on the page, like some boostrap conatiner rather than an alert

Yout-bit replied:
i understand this might be alot, so feel free to leave if it is, if you could just have a very quick look into it

IsaacSo919 replied:
hmmm, i could do that in the future, but it is good for now because im really exhausted lul

Yout-bit replied:
thats fine

Yout-bit reviewed (changes requested):
Overall really great stuff. Would be nice to trim down and remove some of the comments but not required

IsaacSo919 committed:
Carbon foot print will minus, close button added

Carbon foot print will minus, close button added

IsaacSo919 committed:
deleted the redundant line

<script src="path/to/base_game_template.js"></script>

IsaacSo919 committed:
close button added, added the boostrap conatiner

close button added, added the boostrap conatiner, but it is now showing i d k why

IsaacSo919 committed:
added music

added music

Yout-bit commented on src/gseproject/templates/game/tree.html:
```diff
@@ -18,111 +19,222 @@
+<!-- Bootstrap CSS -->
+<link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css">
+<!-- Optional JavaScript; choose one of the two! -->
+<!-- Option 1: jQuery and Bootstrap Bundle (includes Popper) -->
```
please remove this comment, it makes it obvious where it came from

IsaacSo919 replied:
sorted

Yout-bit reviewed (commented):
Tidy the comments please

IsaacSo919 committed:
the sus code is removed

the sus code is removed

IsaacSo919 committed:
trim down comments in tree.html

trim down comments in tree.html

IsaacSo919 committed:
bunch of actions in description

formating +deleted comments in QRcode.html and views.py, check min and max

Yout-bit committed:
views.py tidy

Yout-bit committed:
rename music file
